bulk_grades/api.py

```diff
@@ -331,6 +332,116 @@ def get_rows_to_export(self):
             yield row
 
 
+class InterventionCSVProcessor(CSVProcessor):
+    """
+    CSV Processor for intervention report grades.
+    """
+
+    required_columns = ['user_id', 'course_id']
+
+    def __init__(self, **kwargs):
+        """
+        Create InterventionCSVProcessor.
+        """
+        self.columns = ['user_id', 'username', 'email', 'student_key', 'full_name', 'course_id', 'track', 'cohort',
+                        'number of videos overall', 'number of videos last week', 'number of problems overall',
+                        'number of problems last week',
+                        'number of correct problems overall', 'number of correct problems last week',
+                        'number of problem attempts overall', 'number of problem attempts last week',
+                        'number of forum posts overall', 'number of forum posts last week',
+                        'date last active']
+        self.course_id = None
+        self.track = self.cohort = None
+        super(InterventionCSVProcessor, self).__init__(**kwargs)
+        self._course_key = CourseKey.from_string(self.course_id)
+        self._subsections = self._get_graded_subsections(self._course_key)
+        self.columns.append('course grade letter')
+        self.columns.append('course grade numeric')
+        self._users_seen = set()
+
+    def get_unique_path(self):
+        """
+        Return a unique id for CSVOperations.
+        """
+        return self.course_id
+
+    def _get_graded_subsections(self, course_id):
+        """
+        Return list of graded subsections.
+        """
+        subsections = {}
+        for subsection in grades_api.graded_subsections_for_course_id(course_id):
+            short_block_id = subsection.location.block_id[:8]
+            if short_block_id not in subsections:
+                for key in ('name', 'grade'):
+                    self.columns.append('{}-{}'.format(key, short_block_id))
+                subsections[short_block_id] = (subsection, subsection.display_name)
+        return subsections
+
+    def validate_row(self, row):
+        """
+        Validate row.
+        """
+        super(InterventionCSVProcessor, self).validate_row(row)
+        if row['course_id'] != self.course_id:
+            raise ValidationError(_('Wrong course id {} != {}').format(row['course_id'], self.course_id))
+
+    def get_rows_to_export(self):
+        """
+        Return iterator of rows to export.
+        """
+        enrollments = list(_get_enrollments(self._course_key, track=self.track, cohort=self.cohort))
+        grades_api.prefetch_course_and_subsection_grades(self._course_key, [enroll['user'] for enroll in enrollments])
+        client = LearnerAPIClient(serializer_type='json')
```

Comment:
Since `serializer_type='json'` is the default, you don't need to pass it in here.